circus-train-comparator/src/main/java/com/hotels/bdp/circustrain/comparator/hive/wrappers/PathMetadata.java

```diff
@@ -71,7 +72,11 @@ public int getChecksumLength() {
   }
 
   public byte[] getChecksum() {
-    return checksum;
+    if (checksum == null) {
+      return new byte[] {};
+    } else {
+      return Arrays.copyOf(checksum, checksum.length);
+    }
```

Comment:
This was done because of:
```
com.hotels.bdp.circustrain.comparator.hive.wrappers.PathMetadata.getChecksum() may expose internal 
representation by returning PathMetadata.checksum

Returning a reference to a mutable object value stored in one of the object's fields exposes the internal 
representation of the object. If instances are accessed by untrusted code, and unchecked changes to the mutable 
object would compromise security or other important properties, you will need to do something different. Returning 
a new copy of the object is better approach in many situations.
```